__tests__/app.test.js

```diff
@@ -1,87 +1,158 @@
+      });
+  });
+});
 
+describe("GET /api/topics", () => {
+  test("200: returns a 200 status code", () => {
+    return request(app).get("/api/topics").expect(200);
+  });
+  test("200: returns a list of topics, with the same length as in test data", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.topics.length).toBe(3);
+      });
+  });
+  test("200: returns a list of topics that match the test data", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then((response) => {
+        const topics = response.body.topics;
+        topics.forEach((topic) => {
+          expect(topic).toMatchObject({
+            description: expect.any(String),
+            slug: expect.any(String),
+          });
+        });
+      });
+  });
+});
 
-describe('GET /api/topics', () => {
-    test('200: returns a 200 status code', () => {
-        return request(app)
-        .get('/api/topics')
-        .expect(200)
-    });
-    test('200: returns a list of topics, with the same length as in test data', () => {
-        return request(app)
-        .get('/api/topics')
+describe("GET: /api/articles", () => {
+  test("200: returns a list of articles with all properties from the database", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        const articles = response.body.articles;
+        articles.forEach((article) => {
+          expect(article).toMatchObject({
+            author: expect.any(String),
+            title: expect.any(String),
+            article_id: expect.any(Number),
+            topic: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+          });
+        });
+      });
+  });
+  test("200: returns the correct comment count for each article in the list", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then((response) => {
+        const articles = response.body.articles;
+        // hardcoded from a query to the test database, this won't change as tests are built up
+        const arrayOfCommentCounts = [
+          { article_id: 5, comment_count: "2" },
+          { article_id: 6, comment_count: "1" },
+          { article_id: 1, comment_count: "11" },
+          { article_id: 9, comment_count: "2" },
+          { article_id: 3, comment_count: "2" },
+        ];
+        articles.forEach((article) => {
+          const currentArticleComments = arrayOfCommentCounts.find(
+            (comment_counts) => comment_counts.article_id === article.article_id
+          );
+          if (currentArticleComments) {
+            expect(article.comment_count).toBe(currentArticleComments["comment_count"]);
+          } else {
+            expect(article.comment_count).toBe("0");
+          }
+        });
+      });
```

Comment:
We could have done a comment_count: expect.any(Number) here but this is a nice way to be super accurate 👍